contracts/src/systems/buildings/contracts.cairo

```diff
@@ -40,6 +40,8 @@ mod building_systems {
                 assert!(realm_produces_resource, "realm does not produce specified resource");
             }
 
+            get!(world, entity_id, Owner).assert_caller_owner();
```

Comment:
I just removed this line and place it here, otherwise create couldn't be called for the mines because they have no owner when creating the structure since the mercenaries upgrade https://github.com/BibliothecaDAO/eternum/blob/bd5dbb14c21c9e2dea8f1164a06a1824aed016ca/contracts/src/models/buildings.cairo#L494 I can change this line to use `EntityOwner` though if you think it's better